src/ros_filter.cpp

```diff
@@ -2954,29 +3050,11 @@ namespace RobotLocalization
   }
 
   template<typename T>
-  void RosFilter<T>::clearExpiredHistory(const double cutOffTime)
+  void RosFilter<T>::saveFilterState(const std::string &associatedTopicName)
   {
-    RF_DEBUG("\n----- RosFilter::clearExpiredHistory -----" <<
-             "\nCutoff time is " << cutOffTime << "\n");
-
-    int poppedMeasurements = 0;
-    int poppedStates = 0;
-
-    while (!measurementHistory_.empty() && measurementHistory_.front()->time_ < cutOffTime)
-    {
-      measurementHistory_.pop_front();
-      poppedMeasurements++;
-    }
-
-    while (!filterStateHistory_.empty() && filterStateHistory_.front()->lastMeasurementTime_ < cutOffTime)
-    {
-      filterStateHistory_.pop_front();
-      poppedStates++;
-    }
-
-    RF_DEBUG("\nPopped " << poppedMeasurements << " measurements and " <<
-             poppedStates << " states from their respective queues." <<
-             "\n---- /RosFilter::clearExpiredHistory ----\n" );
+    filterStateHistory_.push_back(generateFilterState());
```

Comment:
Shouldn't this take `associatedTopicName` as an arg? Otherwise it shouldn't be an arg to `saveFilterState()`.

Reply:
Yes, TY.